File: Web/Applications/sportopia.com/ClientApp/app/System/STA/Data/Models/Person.ts
import * as System from "../../../_include";
import * as STA from "../../../STA/_include";
import * as Base from "../Model";

export class Person extends Base.Model {
	constructor(assembly: STA.Context, init?: any) {
		super(assembly, init);
	}

	private __name: string;
	public get Name(): string {
		return this.__name;		
	}
	public set Name(value: string) {
		this.__name = value;
	}

	public static readonly _type: any = {
		"Name": "STA.Data.Models.Person",
		"BaseType": "STA.Data.Model",
		"Properties": [
			{
				"Name": "Name",
				"PropertyType": "string",
				"Variable": "__name"
			}
		]
	};
	public GetType(): any {
		return System.Type.GetType(STA.Data.Models.Person._type.Name);
	}
}

File: Windows/Libraries/STA/Data/Validators/Person.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Reflection;
using System.Text;
using System.Threading.Tasks;
using LMS.Data;

namespace STA.Data.Validators {
    public class Person : LMS.Data.Validator<Context, Models.Person> {
        public Person(LMS.Data.Controller<Context, Models.Person> controller) : base(controller) { }

        public override LMS.Data.Message Validate(PropertyInfo propertyInfo) {
            switch (propertyInfo.Name) {
                case "FirstName":
                    if (string.IsNullOrWhiteSpace(this.Model.FirstName))
                        return LMS.Data.Message.Invalid("Required");
                    break;
                case "LastName":
                    if (string.IsNullOrWhiteSpace(this.Model.LastName))
                        return LMS.Data.Message.Invalid("Reuqired");
                    break;
                default:
                    return base.Validate(propertyInfo);
            }
            return base.Validate(propertyInfo);
        }
    }
}
